### PR TITLE
Add Snapshots to Godotto

### PR DESCRIPTION
**Pull request by xmudrii**
this should add snapshots to godotto

it's wip!
tests are missing and maybe i miss few things, i'm too tired at this point 💤 

@aybabtme what do I need to do to make it appear? i don't any snapshots command in repl 😞

### CODE REVIEW

**Comment by aybabtme** [comment]
hey thanks for this! check how @iheanyi did in https://github.com/aybabtme/godotto/pull/8/files

**Comment by xmudrii** [comment]
@aybabtme thanks for linking this.
I think right now it misses only tests for `pkg/snapshots/snapshots.go`.

if you want to test it/review it, feel free to do so, or wait me to finish tests. 😄

**Comment by xmudrii** [comment]
@aybabtme can you please take a look at this, if you have time ofc? I've got into problem that I don't see.
why does `TestSnapshotsListDroplet` fail? it seems it should work :/

File: pkg/extra/do/mockcloud/client.go [hunk 1]
```diff
@@ -830,3 +834,53 @@ func (mock *MockLoadBalancers) RemoveForwardingRules(ctx context.Context, lbId s
...
+
+func (mock *MockSnapshots) List(ctx context.Context) (<-chan snapshots.Snapshot, <-chan error) {
+	if mock.ListFn != nil {
+		return mock.ListFn(ctx)
+	}
+
+	return mock.wrap.Snapshots().List(ctx)
+}
+func (mock *MockSnapshots) ListDroplet(ctx context.Context) (<-chan snapshots.Snapshot, <-chan error) {
+	if mock.ListFn != nil {
```
Comment (iheanyi) [inline pkg/extra/do/mockcloud/client.go]: I figured out why your `TestSnapshotListDroplet` function is failing. You have a typo here. It should be `mock.ListDropletFn != nil`

File: pkg/extra/do/mockcloud/client.go [hunk 2]
```diff
@@ -830,3 +834,53 @@ func (mock *MockLoadBalancers) RemoveForwardingRules(ctx context.Context, lbId s
...
+		return mock.ListDropletFn(ctx)
+	}
+
+	return mock.wrap.Snapshots().ListDroplet(ctx)
+}
+
+func (mock *MockSnapshots) ListVolume(ctx context.Context) (<-chan snapshots.Snapshot, <-chan error) {
+	if mock.ListFn != nil {
```
Comment (iheanyi) [inline pkg/extra/do/mockcloud/client.go]: This should be `mock.ListVolumeFn != nil` as well.

File: pkg/snapshots/snapshots.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+		svc: client.Snapshots(),
+	}
+
+	for _, applier := range []struct {
+		Name   string
+		Method interface{}
+	}{
+		{"get", svc.get},
+		{"list", svc.list},
+		{"listDroplet", svc.listDroplet},
```
Comment (iheanyi) [inline pkg/snapshots/snapshots.go]: For the sake of uniformity, could this be changed into `list_droplet` please?

File: pkg/snapshots/snapshots.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+		{"listVolume", svc.listVolume},
```
Comment (iheanyi) [inline pkg/snapshots/snapshots.go]: For uniformity, could we change this to `"list_volume"` please? Just the string, not the function definition itself. :)

**Comment by iheanyi** [comment]
@xmudrii Sorry for the late response dude, I know that @aybabtme is busy so I've started taking a look at this PR. I've left some comments with regards to some typos in the code, so those tests should start passing again after you change some things.

**Comment by xmudrii** [comment]
@aybabtme @iheanyi this should be it :D

**Comment by aybabtme** [comment]
🎉 LGTM!

**Pull request merged**